network/p2p/readConnection.go

```diff
@@ -67,31 +67,54 @@ func (rc *readConnection) receiveLoop(wg *sync.WaitGroup) {
 		}
 
 		var msg message.Message
-		// read the nex message (blocking call)
+		// read the next message (blocking call)
 		err := r.ReadMsg(&msg)
 
 		// error handling done similar to comm.go in pubsub (as suggested by libp2p folks)
 		if err != nil {
 			// if the sender closes the connection an EOF is received otherwise an actual error is received
 			if err != io.EOF {
 				rc.log.Error().Err(err)
-				err = rc.stream.Reset()
-				if err != nil {
-					rc.log.Error().Err(err)
-				}
+				rc.resetStream()
 			} else {
-				err = rc.stream.Close()
-				if err != nil {
-					rc.log.Error().Err(err)
-				}
+				rc.closeStream()
 			}
 			return
 		}
 
+		// check message size
+		maxSize := unicastMaxMsgSize(&msg)
+		if msg.Size() > maxSize {
+			// if message size exceeded, close stream and log error
+			rc.closeStream()
+			rc.log.Error().
```

Comment:
I would suggest escalating its level to `fatal`. If such a situation happens we'd better catch it in the snap of a finger than delving into the `error` level logs.

Reply:
i believe `level.fatal` tends to also throw a fatal error and bring down the node, doesn't it?

Reply:
I experimented with a `log.fatal` somewhere else in the code and it does stops the node. Hence, for now I am keeping it as an `error`.